### PR TITLE
added all features

### PR DESCRIPTION
**Pull request by vikash-kanauja**

### CODE REVIEW

File: src/component/AddTodoModal.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  handleInputChange,
+  handleChangeTime,
+  addOrUpdateTodo,
+  time,
+}) => {
+  // Minimum value for the date-time picker
+  const dateTimePickerMinValue = moment().format("YYYY-MM-DDTHH:mm");
+  return (
+    <>
+      {show && (              // Render the modal if `show` is true
```
Comment (yuvrajKaexpertise) [inline src/component/AddTodoModal.js]: show??? what are you showing here?

File: src/component/DeleteTodoModal.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+
+const DeleteTodoModal = ({show,deleteFun ,hideDeletePopModal,id}) => {
```
Comment (yuvrajKaexpertise) [inline src/component/DeleteTodoModal.js]: take a state variable name meaningful;

File: src/component/AddTodoModal.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import moment from "moment";
+
+const AddTodoModal = ({
+  show,
+  todoInputText,
+  dateError,
+  inputError,
+  hideTodoPopup,
```
Comment (yuvrajKaexpertise) [inline src/component/AddTodoModal.js]: state variable name.

File: src/component/TodoItem.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        </div>
+
+        <div className='flex justify-center items-center gap-2 w-[21%] '>
+         <div className={`rounded-full dot w-3 h-3 ${moment(todo.time).isBefore(moment(), "minute")
+            ? "bg-red-500"
+            : todo.completed
+              ? "bg-green-500"
+              : "bg-purple-500"
+            }`}></div>
+          <div className='cursor-pointer	' onClick={() => deletePopup(todo.id)}><MdDelete className='text-red-500' /></div>
```
Comment (yuvrajKaexpertise) [inline src/component/TodoItem.js]: space between quotes???

File: src/component/Todos.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+import AddTodoModal from './AddTodoModal'
+import DeleteTodoModal from './DeleteTodoModal';
+const Todos = () => {
+  const [todoList, setTodoList] = useState([]);
+  const [isShowTodoPopup, setIsShowTodoPopup] = useState(false);
+  const [isShowDeleteModal, setIsShowDeleteModal] = useState(false); // State for showing/hiding delete todo modal
+  const [todoInputText, setTodoInputText] = useState("");
+  const [inputError, setInputError] = useState(false);
+  const [dateError,setDateError] = useState(false)
+  const [deleteTodoId, setDeleteTodoId] = useState(null);
```
Comment (yuvrajKaexpertise) [inline src/component/Todos.js:18]: why are you creating a state variable for deleteId.

File: src/component/Todos.js [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+  const [currentDateAndTime, setCurrentDateAndTime] = useState(moment().format('YYYY-MM-DDTHH:mm'))   // State for current date and time
+  const [time, setTime] = useState(moment().format('YYYY-MM-DDTHH:mm'))
+
+ // Function to show add todo modal
+  const showTodoPopup = () => {
+    setIsShowTodoPopup(true);
+    setTime(moment().format('YYYY-MM-DDTHH:mm'))  // Reset time when showing popup
+  }
+   // Function to hide add todo modal
+  const hideTodoPopup = () => {
```
Comment (yuvrajKaexpertise) [inline src/component/Todos.js]: variable name??

File: src/component/Todos.js [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+    setIsShowTodoPopup(false);
+    setTime(moment().format('YYYY-MM-DDTHH:mm'))
+    setTodoInputText("")
+  }
+
+    // Function to show delete todo modal
+  const showDeletePopupModal = (id) => {
```
Comment (yuvrajKaexpertise) [inline src/component/Todos.js]: You can also delete popup handle in one function don't create two seprate function for this.

File: src/component/Todos.js [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+        },
+        ...prevlist
+      ])
+      setTodoInputText("");
+      hideTodoPopup();
+      setTime(moment().format('YYYY-MM-DDTHH:mm'))
+    }
+  }
+   // Function to delete a todo item
+  const deleteFun = (id) => {
```
Comment (yuvrajKaexpertise) [inline src/component/Todos.js]: deleteFun take a variable name meaningful

File: src/component/Todos.js [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+        },
+        ...prevlist,
+      ]);
+      setTodoInputText("");
+      toggleTodoPopup();
+      setTime(moment().format("YYYY-MM-DDTHH:mm"));
+    }
+  };
+  // Function to delete a todo item
+  const deleteTodo = (id) => {
```
Comment (amrit56789) [inline src/component/Todos.js:128]: instead of passing id you can whole todo like you are passing in edit.

File: src/component/Todos.js [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+    setTodoList(todoList.filter((todo) => todo.id !== id));
+    setDeleteTodoId(null);
```
Comment (amrit56789) [inline src/component/Todos.js:130]: same question again why are you creating a deleteTodoId state variable.

File: src/component/DeleteTodoModal.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+const DeleteTodoModal = ({ showOrHideDeleteModal, deleteTodo, toggleDeleteModal, id }) => {
```
Comment (meanstack9716) [inline src/component/DeleteTodoModal.js:2]: also you are selected todo and id why?

Comment (amrit56789) [inline src/component/DeleteTodoModal.js]: also you are sending deleteTodo, and id. why?

File: src/component/AddTodoModal.js [hunk 12]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import moment from "moment";
+
+const AddTodoModal = ({
+  showOrHideTodoModal,
+  todoInputText,
+  handleInput,
+  error,
+  TodoPopupModal,
```
Comment (yuvrajKaexpertise) [inline src/component/AddTodoModal.js]: TodoPopupModal ?
what is this for? Is it a function
if yes, change the name and start with small case @vikash-kanauja

File: src/component/AddTodoModal.js [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+  addOrUpdateTodo,
+  time,
+}) => {
+  // Minimum value for the date-time picker
+  const dateTimePickerMinValue = moment().format("YYYY-MM-DDTHH:mm");
+  return (
+    <>
```
Comment (yuvrajKaexpertise) [inline src/component/AddTodoModal.js]: why do you need a blank element? @vikash-kanauja

File: src/component/DeleteTodoModal.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+const DeleteTodoModal = ({ showOrHideDeleteModal, deleteTodo, DeletePopupModal, id }) => {
+  return (
+    <>
```
Comment (yuvrajKaexpertise) [inline src/component/DeleteTodoModal.js]: same here @vikash-kanauja

File: src/component/DeleteTodoModal.js [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+      {
+        showOrHideDeleteModal && (
+          <div className="absolute text-center mx-auto w-[80%] top-[30%] left-[10%] bg-white border-2 px-2 py-4 rounded ">
+            <p className="text-2xl text-left w-[80%] mx-auto mb-6">Delete</p>
+            <p className="text-base text-left w-[80%] mx-auto mb-6">
+              Are you sure you want to delete this item ?
+            </p>
+            <div className="flex justify-between w-[80%] mx-auto text-lg">
+              <button className="px-4 py-1 border rounde" onClick={() => DeletePopupModal()}> Cancel</button>
```
Comment (yuvrajKaexpertise) [inline src/component/DeleteTodoModal.js]: use good function name, 2nd function name always start with smallcase @vikash-kanauja

**Review by yuvrajKaexpertise — CHANGES_REQUESTED** [review]

**Pull request merged**